server/controllers/userController.js

```diff
@@ -321,6 +326,64 @@ class UserController {
       util.errorStatus(res, 500, error.name);
     }
   }
+
+  static async getProfile(req, res) {
+    const { is_own_profile } = req;
+    const { id } = req.query;
+
+    const returnUserObj = (userID) => {
+      return models.Users.findByPk(userID, {
+        attributes: ['profilePic', 'bio', 'favoriteQuote', 'favoriteBooks', 'firstName', 'username', 'lastName']
+      })
+    }
+ 
+     try {
+          const user = await returnUserObj(id);
+           return util.successStatus(res, 200, 'profile retrieved successfully', {
+              is_own_profile,
+              firstName: user.firstName,
+              lastName: user.lastName,
+              username: user.username,
+              bio: user.bio,
+              profilePic: user.profilePic,
+              favoriteQuote: user.favoriteQuote,
+              favoriteBooks: user.favoriteBooks
+            })
+
+     } catch(err) {
+      return util.errorStatus(res, 500, err.name)
+     }
+  }
+
+  static async editProfile(req, res) {
+
+    const {...args} = req.body;
+
+    if(Object.keys(args).length <= 0) return util.errorStatus(res, 400, 'You must one or more profile attributes to update')
+
+    const { is_own_profile, loggedinUser } = req;
+
+    if(!is_own_profile) return util.errorStatus(res, 401, 'Unauthorized user');
```

Comment:
WHy was this not done from the middleware?

Reply:
Because I didn't see the need to, the middle ware was only designed to set the value of `is_own_profile` and i only would do that once.